### PR TITLE
Gen 8 moves batch 1

### PR DESCRIPTION
**Pull request by BuffelSaft**
Partially supersedes #1290. Full credit to Nunuchu42 and cfmnephrite for 99% of this PR; I simply updated it, made the requested style changes and fixed a few minor issues I found while testing. Opening this PR as the existing one has been awaiting changes for the better part of a year.

I've thoroughly tested all of these move effects and they're all working very well, if not perfectly.

## Description
Implements the following gen 8move effects:
- EFFECT_JAW_LOCK
- EFFECT_NO_RETREAT
- EFFECT_TAR_SHOT
- EFFECT_POLTERGEIST
- EFFECT_OCTOLOCK
- EFFECT_CLANGOROUS_SOUL
- EFFECT_BOLT_BEAK

Note that I excluded Dragon Darts from this as that's likely to be complex enough for its own PR later.

## Testing notes
Jaw Lock - no issues
No Retreat - no issues
Magic Powder - no issues
Tar Shot - added failure condition if move will have no effect
Poltergeist - no issues
Octolock - Fixed stat drop animation not playing if only sp. def drops. Also has no end of turn animation in Sword, just drops stats, so I removed the wrap animation from it.
Clangorous Soul - no issues
Bolt Beak - no issues

## **Discord contact info**
Buffel Saft#2205

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -9019,6 +9036,88 @@ static void Cmd_various(void)
...
+        }
+        else
+        {
+            PREPARE_ITEM_BUFFER(gBattleTextBuff1, gBattleMons[gActiveBattler].item);
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    case VARIOUS_TRY_NO_RETREAT:
+        if (gDisableStructs[gActiveBattler].noRetreat)
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9093]: braces needed to match pret styling

File: src/battle_script_commands.c [hunk 2]
```diff
@@ -9019,6 +9036,88 @@ static void Cmd_various(void)
...
+        else
+        {
+            if (!(gBattleMons[gActiveBattler].status2 & STATUS2_ESCAPE_PREVENTION))
+                gDisableStructs[gActiveBattler].noRetreat = 1;
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9097]: `gDisableStructs[gActiveBattler].noRetreat = TRUE`

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -9019,6 +9036,88 @@ static void Cmd_various(void)
...
+        }
+        return;
+    case VARIOUS_TRY_TAR_SHOT:
+        if (gDisableStructs[gActiveBattler].tarShot)
+        {
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+        }
+        else
+        {
+            gDisableStructs[gActiveBattler].tarShot = 1;
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9108]: `gDisableStructs[gActiveBattler].tarShot = TRUE`

File: src/battle_script_commands.c [hunk 4]
```diff
@@ -9019,6 +9036,88 @@ static void Cmd_various(void)
...
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    case VARIOUS_CAN_TAR_SHOT_WORK:
+        // Tar Shot will fail if it's already been used on the target and its speed can't be lowered further
+        if (gDisableStructs[gActiveBattler].tarShot 
+            && (gBattleMons[gActiveBattler].statStages[STAT_SPEED] == MIN_STAT_STAGE
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9115]: This can be replaced with `CompareStat(gActivebattler, STAT_SPEED, MIN_STAT_STAGE, CMP_GREATER_THAN)`

File: src/battle_util.c [hunk 5]
```diff
@@ -8820,6 +8837,8 @@ u16 CalcTypeEffectivenessMultiplier(u16 move, u8 moveType, u8 battlerAtk, u8 bat
         modifier = CalcTypeEffectivenessMultiplierInternal(move, moveType, battlerAtk, battlerDef, recordAbilities, modifier);
         if (gBattleMoves[move].effect == EFFECT_TWO_TYPED_MOVE)
             modifier = CalcTypeEffectivenessMultiplierInternal(move, gBattleMoves[move].argument, battlerAtk, battlerDef, recordAbilities, modifier);
+        else if (moveType == TYPE_FIRE && gDisableStructs[battlerDef].tarShot)
```
Comment (ghoulslash) [inline src/battle_util.c:8840]: would this be better in `MulByTypeEffectiveness`?

File: src/battle_script_commands.c [hunk 6]
```diff
@@ -9019,6 +9036,88 @@ static void Cmd_various(void)
...
+            gDisableStructs[gActiveBattler].octolock = 1;
+            gBattleMons[gActiveBattler].status2 |= STATUS2_ESCAPE_PREVENTION;
+            gDisableStructs[gActiveBattler].battlerPreventingEscape = gBattlerAttacker;
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    case VARIOUS_CHECK_POLTERGEIST:
+        if (gBattleMons[gActiveBattler].item == ITEM_NONE
+           || gFieldStatuses & STATUS_FIELD_MAGIC_ROOM
+           || GetBattlerAbility(gActiveBattler) == ABILITY_KLUTZ)
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9081]: Klutz should check gastro acid / neutralizing gas.

Alternatively use `GetBattlerHoldEffect` and make the `checkNegating` argument a bitfield of flags, where all TRUE would be 0xFF, and this specific check would be all flags except one for embargo.

File: src/battle_script_commands.c [hunk 7]
```diff
@@ -9019,6 +9036,88 @@ static void Cmd_various(void)
...
+        }
+        return;
+    case VARIOUS_SET_OCTOLOCK:
+        if (gDisableStructs[gActiveBattler].octolock)
+        {
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+        }
+        else
+        {
+            gDisableStructs[gActiveBattler].octolock = 1;
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9072]: `gDisableStructs[gActiveBattler].octolock = TRUE;`

File: data/battle_scripts_1.s [hunk 8]
```diff
@@ -401,6 +408,111 @@ BattleScript_EffectAuraWheel: @ Aura Wheel can only be used by Morpeko
...
+	waitmessage B_WAIT_TIME_LONG
+	goto BattleScript_MoveEnd
+
+BattleScript_OctolockEndTurn::
+	jumpifstat BS_TARGET, CMP_GREATER_THAN, STAT_DEF, MIN_STAT_STAGE, BattleScript_OctolockLowerDef
+	jumpifstat BS_TARGET, CMP_GREATER_THAN, STAT_SPDEF, MIN_STAT_STAGE, BattleScript_OctolockTryLowerSpDef
+	goto BattleScript_OctolockEnd2
+BattleScript_OctolockLowerDef:
+	playstatchangeanimation BS_ATTACKER, BIT_DEF | BIT_SPDEF, STAT_CHANGE_NEGATIVE
+	jumpifability BS_TARGET, ABILITY_BIG_PECKS, BattleScript_OctolockTryLowerSpDef
```
Comment (ghoulslash) [inline data/battle_scripts_1.s:443]: shouldn't this check be before `playstatchangeanimation`? alternatively, it should be checked in `ChangeStatBuffs` similar to hyper cutter, but that should probably be a separate PR

File: data/battle_scripts_1.s [hunk 9]
```diff
@@ -401,6 +408,111 @@ BattleScript_EffectAuraWheel: @ Aura Wheel can only be used by Morpeko
...
+	setstatchanger STAT_DEF, 1, TRUE
+	statbuffchange STAT_BUFF_ALLOW_PTR, BattleScript_OctolockTryLowerSpDef
```
Comment (ghoulslash) [inline data/battle_scripts_1.s:445]: these may need to include `MOVE_EFFECT_AFFECTS_USER` or be sure `gBattlerAttacker = gBattlerTarget` to prevent Mirror Armor from triggering.

File: data/battle_scripts_1.s [hunk 10]
```diff
@@ -401,6 +408,111 @@ BattleScript_EffectAuraWheel: @ Aura Wheel can only be used by Morpeko
 	waitmessage B_WAIT_TIME_LONG
 	goto BattleScript_MoveEnd
 
+BattleScript_EffectClangorousSoul:
+	attackcanceler
+	attackstring
+	ppreduce
+	cutonethirdhpraisestats BattleScript_ButItFailed
+	orword gHitMarker, HITMARKER_IGNORE_SUBSTITUTE | HITMARKER_IGNORE_DISGUISE
```
Comment (ghoulslash) [inline data/battle_scripts_1.s:416]: should include `HITMARKER_SKIP_DMG_TRACK` and `HITMARKER_PASSIVE_DAMAGE`

File: src/battle_script_commands.c [hunk 11]
```diff
@@ -9019,6 +9036,88 @@ static void Cmd_various(void)
...
         break;
+    case VARIOUS_CUT_1_3_HP_RAISE_STATS:
+        {
+            bool8 atLeastOneStatBoosted = FALSE;
+            bool8 hasContrary = (GetBattlerAbility(gBattlerAttacker) == ABILITY_CONTRARY);
+            u16 hpFraction = max(1, gBattleMons[gBattlerAttacker].maxHP / 3);
+
+            for (i = 1; i < NUM_STATS; i++)
+            {
+                if (!(gBattleMons[gBattlerAttacker].statStages[i] == MAX_STAT_STAGE
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9047]: Can use `CompareStat(gBattlerAttacker, i, MAX_STAT_STAGE, CMP_LESS_THAN)`

File: src/battle_script_commands.c [hunk 12]
```diff
@@ -9019,6 +9036,88 @@ static void Cmd_various(void)
...
+                      || (hasContrary && gBattleMons[gBattlerAttacker].statStages[i] == MIN_STAT_STAGE)))
+                {
+                    atLeastOneStatBoosted = TRUE;
+                    break;
+                }
+            }
+            if (atLeastOneStatBoosted && gBattleMons[gBattlerAttacker].hp > hpFraction)
+            {
+                gBattleMoveDamage = hpFraction;
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9056]: shouldn't this be negative?

**Review by ghoulslash — CHANGES_REQUESTED** [review]

File: src/battle_script_commands.c (hunk 12 above)
Comment (BuffelSaft) [reply src/battle_script_commands.c:9216]: No, negative damage restores HP (used for giga drain etc.)

File: src/battle_script_commands.c (hunk 6 above)
Comment (BuffelSaft) [reply src/battle_script_commands.c:9241]: Doesn't `GetBattlerAbility` already handle gastro acid and neutralizing gas?

File: src/battle_util.c (hunk 5 above)
Comment (BuffelSaft) [reply src/battle_util.c]: Yeah that would be better!

**Comment by BuffelSaft** [comment]
Ready for re-review!

File: data/battle_scripts_1.s [hunk 13]
```diff
@@ -471,16 +471,16 @@ BattleScript_OctolockEndTurn::
 	jumpifstat BS_TARGET, CMP_GREATER_THAN, STAT_SPDEF, MIN_STAT_STAGE, BattleScript_OctolockTryLowerSpDef
 	goto BattleScript_OctolockEnd2
 BattleScript_OctolockLowerDef:
-	playstatchangeanimation BS_ATTACKER, BIT_DEF | BIT_SPDEF, STAT_CHANGE_NEGATIVE
 	jumpifability BS_TARGET, ABILITY_BIG_PECKS, BattleScript_OctolockTryLowerSpDef
```
Comment (ghoulslash) [inline data/battle_scripts_1.s:474]: big pecks is now lowering just sp def, but not playing any stat change animation. You can set `setbyte sSTAT_ANIM_PLAYED, TRUE` after the first one, and then check if this byte is set to decide whether or not to play just the defense-lowering anim

File: src/battle_script_commands.c (hunk 12 above)
Comment (ghoulslash) [reply src/battle_script_commands.c:9216]: wow I am dumb. you're right

File: src/battle_script_commands.c (hunk 6 above)
Comment (ghoulslash) [reply src/battle_script_commands.c:9241]: wow I must have had a brain aneurysm when reviewing this... you're right

**Review by ghoulslash — CHANGES_REQUESTED** [review]

File: data/battle_scripts_1.s (hunk 13 above)
Comment (BuffelSaft) [reply data/battle_scripts_1.s:475]: Done! Ready for re-re-review!

**Review by ghoulslash — APPROVED** [review]

**Comment by cfmnephrite** [comment]
Thanks for doing this, really appreciate it!
